storage: microbenchmark cached data versioning

```
--- FAIL: BenchmarkMVCCScan/rows=100/versions=2/valueSize=64/numRangeKeys=1/headers=false
    test_fixtures.go:85: using existing fixture "mvcc-fmtver_17-numKeys_100000-numVersions_2-valueBytes_64-numColFams_0-numRangeKeys_1" in "/Users/jackson/Library/Caches/crdb-test-fixtures/mvcc-fmtver_17-numKeys_100000-numVersions_2-valueBytes_64-numColFams_0-numRangeKeys_1"
    bench_data_test.go:101: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/storage/bench_data_test.go:101
        	            				github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:1122
        	            				github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:647
        	            				GOROOT/src/testing/benchmark.go:193
        	            				GOROOT/src/testing/benchmark.go:215
        	            				src/runtime/asm_arm64.s:1222
        	Error:      	Received unexpected error:
        	            	store last used with cockroach non-development version v23.2 cannot be opened by development version v1000024.2-upgrading-to-1000024.3-step-022
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.newPebble
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/pebble.go:1415
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.Open
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/open.go:543
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.getInitialStateEngine
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/bench_data_test.go:100
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.runMVCCScan
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:1122
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.BenchmarkMVCCScan.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:647
        	            	  | testing.(*B).runN
        	            	  | 	GOROOT/src/testing/benchmark.go:193
        	            	  | testing.(*B).run1.func1
        	            	  | 	GOROOT/src/testing/benchmark.go:215
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (2) store last used with cockroach non-development version v23.2 cannot be opened by development version v1000024.2-upgrading-to-1000024.3-step-022
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	BenchmarkMVCCScan/rows=100/versions=2/valueSize=64/numRangeKeys=1/headers=false
--- FAIL: BenchmarkMVCCScan/rows=1000/versions=10/valueSize=64/numRangeKeys=100/headers=false
    test_fixtures.go:85: using existing fixture "mvcc-fmtver_17-numKeys_100000-numVersions_10-valueBytes_64-numColFams_0-numRangeKeys_100" in "/Users/jackson/Library/Caches/crdb-test-fixtures/mvcc-fmtver_17-numKeys_100000-numVersions_10-valueBytes_64-numColFams_0-numRangeKeys_100"
    bench_data_test.go:101: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/storage/bench_data_test.go:101
        	            				github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:1122
        	            				github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:647
        	            				GOROOT/src/testing/benchmark.go:193
        	            				GOROOT/src/testing/benchmark.go:215
        	            				src/runtime/asm_arm64.s:1222
        	Error:      	Received unexpected error:
        	            	store last used with cockroach non-development version v23.2 cannot be opened by development version v1000024.2-upgrading-to-1000024.3-step-022
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.newPebble
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/pebble.go:1415
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.Open
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/open.go:543
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.getInitialStateEngine
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/bench_data_test.go:100
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.runMVCCScan
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:1122
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.BenchmarkMVCCScan.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/bench_test.go:647
        	            	  | testing.(*B).runN
        	            	  | 	GOROOT/src/testing/benchmark.go:193
        	            	  | testing.(*B).run1.func1
        	            	  | 	GOROOT/src/testing/benchmark.go:215
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (2) store last used with cockroach non-development version v23.2 cannot be opened by development version v1000024.2-upgrading-to-1000024.3-step-022
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	BenchmarkMVCCScan/rows=1000/versions=10/valueSize=64/numRangeKeys=100/headers=false
BenchmarkMVCCScan/rows=1000/versions=2/valueSize=64/numRangeKeys=0/headers=true      	       1	    333250 ns/op	 192.05 MB/s
--- BENCH: BenchmarkMVCCScan/rows=1000/versions=2/valueSize=64/numRangeKeys=0/headers=true
    test_fixtures.go:85: using existing fixture "mvcc-fmtver_17-numKeys_100000-numVersions_2-valueBytes_64-numColFams_0-numRangeKeys_0-headers_true" in "/Users/jackson/Library/Caches/crdb-test-fixtures/mvcc-fmtver_17-numKeys_100000-numVersions_2-valueBytes_64-numColFams_0-numRangeKeys_0-headers_true"
test logs left over in: /tmp/cockroach/_tmp/d1091de82354dd1e885605d51bd315be/logBenchmarkMVCCScan2526721491
--- FAIL: BenchmarkMVCCScan

 ```

Jira issue: CRDB-43564

--- a/pkg/storage/bench_data_test.go
+++ b/pkg/storage/bench_data_test.go
@@ -210,7 +210,7 @@ var _ initialState = mvccBenchData{}
 func (d mvccBenchData) Key() []string {
 	key := []string{
 		"mvcc",
-		fmt.Sprintf("fmtver_%d", previousReleaseFormatMajorVersion),
+		fmt.Sprintf("cv_%s", clusterversion.PreviousRelease.Version()),
 		fmt.Sprintf("numKeys_%d", d.numKeys),
 		fmt.Sprintf("numVersions_%d", d.numVersions),
 		fmt.Sprintf("valueBytes_%d", d.valueBytes),
@@ -428,7 +428,7 @@ var _ initialState = mvccImportedData{}
 func (i mvccImportedData) Key() []string {
 	key := []string{
 		"mvcc",
-		fmt.Sprintf("fmtver_%d", previousReleaseFormatMajorVersion),
+		fmt.Sprintf("cv_%s", clusterversion.PreviousRelease.Version()),
 		fmt.Sprintf("streak_%d", i.streakBound),
 		fmt.Sprintf("keys_%d", i.keyCount),
 		fmt.Sprintf("valueBytes_%d", i.valueBytes),

